sasdata/temp_hdf5_reader.py

```diff
@@ -0,0 +1,154 @@
+
+        attributes = {name: hdf5_entry.attrs[name] for name in hdf5_entry.attrs}
+
+        if isinstance(hdf5_entry.dtype, np.dtypes.BytesDType):
+            data = hdf5_entry[()][0].decode("utf-8")
+
+            return SASDataDataset[str](
+                name=hdf5_entry.name,
+                data=data,
+                attributes=attributes)
+
+        else:
+            data = np.array(hdf5_entry, dtype=hdf5_entry.dtype)
+
+            return SASDataDataset[np.ndarray](
+                name=hdf5_entry.name,
+                data=data,
+                attributes=attributes)
+
+    elif isinstance(hdf5_entry, HDF5Group):
+        return SASDataGroup(
+            name=hdf5_entry.name,
+            children={key: recurse_hdf5(hdf5_entry[key]) for key in hdf5_entry.keys()})
+
+    else:
+        raise TypeError(f"Unknown type found during HDF5 parsing: {type(hdf5_entry)} ({hdf5_entry})")
+
+GET_UNITS_FROM_ELSEWHERE = units.meters
+def connected_data(node: SASDataGroup, name_prefix="") -> list[NamedQuantity]:
+    """ In the context of NeXus files, load a group of data entries that are organised together
+    match up the units and errors with their values"""
+    # Gather together data with its error terms
+
+    uncertainty_map = {}
+    uncertainties = set()
+    entries = {}
+
+    for name in node.children:
+
+        child = node.children[name]
+
+        if "units" in child.attributes:
+            units = parse(child.attributes["units"])
+        else:
+            units = GET_UNITS_FROM_ELSEWHERE
+
+        quantity = NamedQuantity(name=name_prefix+child.name,
+                                 value=child.data,
+                                 units=units)
+
+        # Turns out people can't be trusted to use the same keys here
+        if "uncertainty" in child.attributes or "uncertainties" in child.attributes:
+            try:
+                uncertainty_name = child.attributes["uncertainty"]
+            except:
+                uncertainty_name = child.attributes["uncertainties"]
+            uncertainty_map[name] = uncertainty_name
+            uncertainties.add(uncertainty_name)
+
+        entries[name] = quantity
+
+    output = []
+
+    for name, entry in entries.items():
+        if name not in uncertainties:
+            if name in uncertainty_map:
+                uncertainty = entries[uncertainty_map[name]]
+                new_entry = entry.with_standard_error(uncertainty)
+                output.append(new_entry)
+            else:
+                output.append(entry)
+
+    return output
+
+
+def load_data(filename) -> list[SasData]:
+    with h5py.File(filename, 'r') as f:
+
+        loaded_data: list[SasData] = []
+
+        for root_key in f.keys():
+
+            entry = f[root_key]
+
+            data_contents = []
+            raw_metadata = {}
+
+            entry_keys = [key for key in entry.keys()]
+
+            if "sasdata" not in entry_keys and "data" not in entry_keys:
```

Comment:
Does this account for the situation where `entry_keys` contains something like `'sasdata01'`?